crates/sncast/src/helpers/fee.rs

```diff
@@ -105,6 +112,17 @@ pub struct FeeSettings {
     pub l2_gas_price: Option<u128>,
     pub l1_data_gas: Option<u64>,
     pub l1_data_gas_price: Option<u128>,
+    pub tip: Option<u64>,
+}
+
+impl FeeSettings {
+    #[must_use]
+    pub fn update_tip(&self, tip: u64) -> FeeSettings {
```

Comment:
I think it would make sense for it to take `self` so it's clearer that we are converting the existing instance to a new one with tip. Don't have a strong opinion on it though.